### PR TITLE
integrate deepeval metric with remote endpoint, like tgi server.

### PR DESCRIPTION
**Pull request by lkk12014402**
## Description

customize model to support tgi/vllm endpoint for evaluation with deepeval metrics

### CODE REVIEW

File: evals/evaluation/deepeval/models/endpoint_models.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+from functools import cached_property
+from typing import List, Optional, Tuple, Union
+
+import requests
+from aiohttp import ClientSession, TCPConnector
+from deepeval.models.gpt_model import GPTModel
+from requests.exceptions import RequestException
+
+
+class TGIEndpointModel(GPTModel):
```
Comment (lvliang-intel) [inline evals/evaluation/deepeval/models/endpoint_models.py:16]: TGI is specific, we can refer to it as 'OpeaServiceModel' as we may also use vLLM as a serving framework.

File: tests/test_answer_relevancy.py [hunk 2]
```diff
@@ -24,6 +24,37 @@ def test_relevancy(self):
         score = metric.measure_zh(test_case)
         print(score)
 
+    @unittest.skip("need pass localhost id")
```
Comment (XinyuYe-Intel) [inline tests/test_answer_relevancy.py:27]: Why skip the test yet write it?

Comment (lkk12014402) [reply tests/test_answer_relevancy.py:27]: need a remote service, but we don't setup now

**Review by XinyuYe-Intel — APPROVED** [review]

File: evals/evaluation/deepeval/models/endpoint_models.py (hunk 1 above)
Comment (lkk12014402) [reply evals/evaluation/deepeval/models/endpoint_models.py:16]: different inference backends have different output format, so I define specific classes for different backends

**Review by lvliang-intel — APPROVED** [review]

**Review by hshen14 — APPROVED** [review]

**Pull request merged**